src/main/java/com/epam/page/object/generator/builders/searchrule/SearchRuleBuilders.java

```diff
@@ -0,0 +1,35 @@
+package com.epam.page.object.generator.builders.searchrule;
+
+import com.epam.page.object.generator.containers.SupportedTypesContainer;
+import com.epam.page.object.generator.models.RawSearchRule;
+import com.epam.page.object.generator.models.searchrule.SearchRule;
+import com.epam.page.object.generator.utils.SearchRuleExtractor;
+import com.epam.page.object.generator.utils.SelectorUtils;
+import com.epam.page.object.generator.utils.XpathToCssTransformer;
+import java.util.Map;
+
+public class SearchRuleBuilders {
```

Comment:
I would suggest to merge 3 classes
- SearchRuleBuilders
- SearchRuleBuildersFactory
- TypeTransformer
into single class SearchRuleFactory

Reply:
Yes, I think that we can merge TypeTransformer and SearchRuleBuilders, but SearchRuleBuildersFactory I think need to stay separately, because if we have a new builder we will be must to add him in this class. This class some kind of config file, where we write which builders we will be used for building searchRules and when we open this file we see only one method where we should add new builder.